liver_imaging_analysis/engine/engine.py

```diff
@@ -358,34 +407,56 @@ def fit(
             Directory to save best weights at. 
             Default is the potential path in config.
         """
+ 
         for epoch in range(epochs):
+
+            forward_time_per_epoch,post_process_time_per_epoch,metric_time_per_epoch,back_prop_time_per_epoch= 0.0,0.0,0.0,0.0
+            pre_train = time.time()
             print(f"\nEpoch {epoch+1}/{epochs}\n-------------------------------")
             training_loss = 0
             training_metric = 0
             self.network.train()
             progress_bar(0, len(self.train_dataloader))  # epoch progress bar
+            epoch_start_timestamps=time.time()
             for batch_num, batch in enumerate(self.train_dataloader):
                 progress_bar(batch_num + 1, len(self.train_dataloader))
                 batch[Keys.IMAGE] = batch[Keys.IMAGE].to(self.device)
                 batch[Keys.LABEL] = batch[Keys.LABEL].to(self.device)
+                pre_forward = time.time()
                 batch[Keys.PRED] = self.network(batch[Keys.IMAGE])
                 loss = self.loss(batch[Keys.PRED], batch[Keys.LABEL])
+                post_forward = time.time()-pre_forward
+                forward_time_per_epoch += post_forward
                 # Apply post processing transforms and calculate metrics
+                pre_post_process_time = time.time()
                 batch = self.post_process(batch)
+                post_post_process_time = time.time() -pre_post_process_time
+                post_process_time_per_epoch += post_post_process_time
+                pre_metric = time.time()
                 self.metrics(batch[Keys.PRED].int(), batch[Keys.LABEL].int())
+                post_metric = time.time()-pre_metric
+                metric_time_per_epoch += post_metric
                 # Backpropagation
+                pre_back_prop_time = time.time()
```

Comment:
Oh no, please don't leave these time calculations by default in the engine. Please remove all of these time debugging

Reply:
done